llvm/test/Transforms/PhaseOrdering/min_max_loop.ll

```diff
@@ -0,0 +1,126 @@
+; RUN: opt < %s --O3 -S | FileCheck %s
+
+define signext i16 @vecreduce_smax_v2i16(i32 noundef %0, ptr noundef %1) #0 {
```

Comment:
Could you please reduce the test case and move it into `InstCombine`?

Reply:
I think the point of these tests is to check interaction between IPSCCP and InstCombine, so a PhaseOrdering test makes sense.

However, in the current form these tests are too hard to understand. I would suggest getting the IR again with `-fno-discard-value-names` and then running `opt -S -passes=sroa` over it to convert it into SSA form.